includes/admin/class-wc-rest-stripe-payment-gateway-controller.php

```diff
@@ -110,7 +113,7 @@ public function update_payment_gateway_settings( WP_REST_Request $request ) {
 			$this->update_is_gateway_enabled( $request );
 			$this->update_gateway_name( $request );
 			$this->update_gateway_description( $request );
-
+			do_action( 'wc_stripe_update_payment_gateway_settings', $request, $id );
```

Comment:
This is working, I see the setting being saved just fine. But I'm wondering whether calling a method for updating the expiration setting would be better than introducing an action for this. We could follow what we're doing for `update_gateway_name()` and those other settings. 

I understand the expiration setting is specific for Boleto, so maybe we could add a check for which gateway is getting updated before trying to update the specific setting. Just throwing ideas in, other approaches could do as well.

Reply:
This is the original way I did it, but I think it's odd to put Boleto logic in this controller, so I moved it out.  

If I understood correctly, this controller handles updating `enabled`, `name`, and `description` which are common on all gateways.  Since expiration is specific to Boleto, I felt we should keep it separate.  Thoughts?

Reply:
Yup, I fully agree on `WC_REST_Stripe_Payment_Gateway_Controller` not handling the details of settings for a specific gateway. I'm thinking of keeping it separate as you did, but using an internal method call instead of an action for it.

For example, we could have a private method in `WC_REST_Stripe_Payment_Gateway_Controller` called `update_gateway_specific_settings` (or anything). Here we could check if the current gateway has a method for updating specific settings, say, a method called `update_scpecific_settings`. If it does, then call that method and the gateway will handle its stuff internally.

The actions we introduce become public interfaces third-party developers could rely on. I think it's better if we can implement this functionality without having to take care of a new hook in the future. I also think it's more readable and easier to test if we can achieve the behavior we want by making internal calls.